### PR TITLE
Add test for failfast in event of corrupted state exception. Add test…

### PR DESCRIPTION
**Pull request by rahku**
… for Assembly.Load by AssemblyName when AssemblyName.CodeBase property is set

### CODE REVIEW

**Comment by rahku** [comment]
PTAL @jkotas @gkhanna79

File: src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public class HandleProcessCorruptedStateExceptionsTests : RemoteExecutorTestBase
+    {
+        [DllImport("kernel32.dll")]
+        static extern void RaiseException(uint dwExceptionCode, uint dwExceptionFlags, uint nNumberOfArguments, IntPtr lpArguments);
+
+        [HandleProcessCorruptedStateExceptions]
+        static void causeAVInNative()
+        {
+            try {
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: Nit: brace on next line

File: src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+using System.Text.RegularExpressions;
+using System.Threading;
+using Xunit;
+
+namespace System.Runtime.ExceptionServices.Tests
+{
+    public class HandleProcessCorruptedStateExceptionsTests : RemoteExecutorTestBase
+    {
+        [DllImport("kernel32.dll")]
+        static extern void RaiseException(uint dwExceptionCode, uint dwExceptionFlags, uint nNumberOfArguments, IntPtr lpArguments);
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: What about Unix?  And is this allowed for UWP or will this need to be factored out for that, too?

File: src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        [HandleProcessCorruptedStateExceptions]
+        static void causeAVInNative()
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: Nit: cause => Cause

File: src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: Ah, I see you only invoke this for Windows, so ignore the Unix question.  But the UWP question remains.

Comment (jkotas) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: RaiseException should work fine on UWP.

Comment (stephentoub) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: Cool

**Comment by jkotas** [comment]
LGTM modulo nits

File: src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            try {
+                RaiseException(0xC0000005, 0, 0, IntPtr.Zero);
```
Comment (gkhanna79) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: How is this AV in native code? Does the AV address computed by the OS to be in OS's native code?

Comment (gkhanna79) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: Are you sure this is not triggering NullRef exception?

Comment (rahku) [inline src/System.Runtime/tests/System/Runtime/ExceptionServices/HandleProcessCorruptedStateExceptions.cs]: Yes address of RaiseException is what ProcessCLRException gets as exception ip. No this does not trigger null ref . I have verified that.

**Pull request merged**